plugins/module_utils/network/nxos/config/ospfv2/ospfv2.py

```diff
@@ -0,0 +1,232 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""
+The nxos_ospfv2 class
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+from copy import deepcopy
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.rm_templates.ospfv2 import (
+    Ospfv2Template,
+)
+from ansible_collections.ansible.netcommon.plugins.module_utils.network.common.utils import (
+    dict_merge,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.nxos.facts.facts import (
+    Facts,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.common.resource_module import (
+    ResourceModule,
+)
+from ansible_collections.cisco.nxos.plugins.module_utils.network.common.rm_utils import (
+    get_from_dict,
+)
+
+
+class Ospfv2(ResourceModule):
+    """
+    The nxos_ospfv2 class
+    """
+
+    def __init__(self, module):
+        super(Ospfv2, self).__init__(
+            empty_fact_val={},
+            facts_module=Facts(module),
+            module=module,
+            resource="ospfv2",
+            tmplt=Ospfv2Template(),
+        )
+        self.parsers = [
```

Comment:
is it required to initialize parsers value in __init__ as not being used by any other fns except _compare, ideally it should be declared under _compare fn. itself

Reply:
I find this approach to be much more cleaner than having a list as big as this within a function. 

Also, the original intention was to reuse a sub-list of the parser list for VRF comparison since not all the global keys are valid for VRF context. That way, we can decrease the run time a little bit by not making unnecessary comparison in base class compare().

Reply:
I am yet to make that change. I'll push that along with the unit tests.